components/encryption/src/manager/mod.rs

```diff
@@ -246,8 +243,13 @@ impl Dicts {
                 Some(file_info) => file_info.clone(),
                 None => {
                     // Could be a plaintext file not tracked by file dictionary.
-                    info!("link untracked plaintext file"; "src" => src_fname, "dst" => dst_fname);
-                    return Ok(());
+                    return Err(Error::Io(IoError::new(
```

Comment:
In that case can we check to see that the pre-existing plaintext file exists on the file system?

Reply:
we always assume the source file exists on the file system when `link_file` is called. If the file entry exists in the file dictionary, we assume the file is encrypted, otherwise unencrypted.

Reply:
It is strange to have to assume so much. I am wondering if this can be improved. However, it seems that some of this behavior is forced by the RocksDB encryption implementation?

Reply:
not really. we make use of some of original rocksdb encryption implementation, but the key manager logic are new. The design of the key manager and the encryption module is that it has no knowledge of the actual file system and only care about encryption metadata. Interaction with file system should be handled in the higher level, i.e. the work-in-progress `file_system` crate.

Reply:
Can the caller check for this case before calling the function? Then we don't need any assumptions.

Reply:
That's not an assumption. The point is the encryption crate should not aware of file system stuff, and just focus on encryption metadata.

Reply:
We talked about this and agreed that in a separate PR we can change this from `Ok(()` to `None` and return an `Option` type.